File: src/AdventureWorks.Application/DataEngine/Production/ProductReview/Commands/Queries/GetProductReviews/ProductReviewLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Production;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;


namespace AdventureWorks.Application.DataEngine.Production.ProductReview.Queries.GetProductReviews
{
    public partial class ProductReviewLookupModel : IHaveCustomMapping
    {
        public int ProductReviewID { get; set; }
        public int ProductID { get; set; }
        public string ReviewerName { get; set; }
        public DateTime ReviewDate { get; set; }
        public string EmailAddress { get; set; }
        public int Rating { get; set; }
        public string Comments { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Production.ProductReview, ProductReviewLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<ProductReviewLookupModel, BaseProductReview>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/production/product-review/product-review-lookup-model.ts

export interface IProductReviewLookupModel {
  productReviewID: number;
  productID: number;
  reviewerName: string;
  reviewDate: Date | string;
  emailAddress: string;
  rating: number;
  comments: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class ProductReviewLookupModel implements IProductReviewLookupModel {
  productReviewID: number;
  productID: number;
  reviewerName: string;
  reviewDate: Date | string;
  emailAddress: string;
  rating: number;
  comments: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<ProductReviewLookupModel>) {
    if (!!init) {
      Object.assign<ProductReviewLookupModel, Partial<ProductReviewLookupModel>>(this, init);
    }
  }
}
